src/osd/OSD.cc

```diff
@@ -10543,14 +10543,8 @@ void OSD::ShardedOpWQ::_process(uint32_t thread_index, heartbeat_handle_d *hb)
     sdata->shard_lock.lock();
 
     auto q = sdata->pg_slots.find(token);
-    if (q == sdata->pg_slots.end()) {
-      // this can happen if we race with pg removal.
```

Comment:
why does this comment not hold true any longer? why can we not race with pg removal?